backend/src/govsg/middlewares/govsg-verification.middleware.ts

```diff
@@ -128,3 +128,25 @@ export const resendPasscodeCreationMessage = async (
     passcode_creation_wamid: passcodeCreationWamid,
   })
 }
+
+export const trackPasscodeReveal = async (
+  req: Request,
+  res: Response
+): Promise<Response | void> => {
+  const { govsg_message_id: govsgMessageId } = req.body
+  const userClickedAt = new Date()
+  await GovsgVerification.update(
+    { userClickedAt },
+    {
+      where: {
+        govsgMessageId,
+        userClickedAt: {
+          [Op.eq]: null,
```

Comment:
Could I ask why this is necessary? From my understanding, we incur a DB call for each invocation of this function and even if we perform the update each time, it still counts as one click in our metrics.

Reply:
Just a personal preference of wanting the value to be the timestamp of the first click ever.